src/datepicker/datepicker.js

```diff
@@ -71,10 +71,12 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
   $scope.uniqueId = 'datepicker-' + $scope.$id + '-' + Math.floor(Math.random() * 10000);
 
   if (angular.isDefined($attrs.initDate)) {
-    this.activeDate = dateParser.fromTimezone($scope.$parent.$eval($attrs.initDate), ngModelOptions.timezone) || new Date();
+    var initDate = dateParser.fromTimezone($scope.$parent.$eval($attrs.initDate), ngModelOptions.timezone);
+    this.activeDate = !isNaN(initDate) ? initDate : new Date();
```

Comment:
All of this with the initDate and setting to this.activeDate can be abstracted out to a common helper function to avoid duplicated code.